src/Extension/HeadingPermalink/HeadingPermalinkProcessor.php

```diff
@@ -57,6 +60,20 @@ public function __invoke(DocumentParsedEvent $e): void
         }
     }
 
+    private function useSlugGeneratorFromConfigurationIfProvided(): void
+    {
+        $generator = $this->config->get('heading_permalink/slug_generator');
+        if ($generator === null) {
+            return;
+        }
```

Comment:
This is a rudimentary example (wrote here) of the static method detection implementation I described above:

```php
<?php

    if (is_string($generator)) {
      // Class was provided, but didn't specify the supported static method.
      // Default to it.
      if (strpos($generator, '::') === FALSE && class_exists($generator)) {
        $class = $generator;
        $method = 'createSlug';
      }
      elseif (strpos($generator, '::') !== FALSE) {
        [$class, $method] = explode('::', $generator);
      }

      if (!is_subclass_of($class, SlugGeneratorInterface::class) || $method !== 'createSlug') {
        throw new InvalidOptionException('The heading_permalink/slug_generator option that supplies a class::createSlug string must implement the interface ' . SlugGeneratorInterface::class);
      }
    }

Reply:
As mentioned above, I don't think it should be the job of this library to instantiate objects for the user - something external to this package should handle that (whether it's a DI container, Drupal module, or something else)

Reply:
That's fine. I can abstract it on the Drupal side of things.